libs/providers/flipt/src/lib/flipt-provider.ts

```diff
@@ -0,0 +1,235 @@
+  TypeMismatchError,
+  ProviderNotReadyError,
+  StandardResolutionReasons,
+  GeneralError,
+} from '@openfeature/server-sdk';
+import { transformContext } from './context-transformer';
+
+export interface FliptClientParameters {
+  url: string;
+  authenticationStrategy?: AuthenticationStrategy;
+}
+
+export class FliptProvider implements Provider {
+  readonly metadata: ProviderMetadata = {
+    name: 'flipt-client-provider',
+  };
+
+  private _clientParameters: FliptClientParameters;
+  private _namespace: string;
+  private _client?: FliptClient;
+
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  constructor(namespace: string, clientParameters: FliptClientParameters) {
+    this._clientParameters = clientParameters;
+    this._namespace = namespace;
+  }
+
+  private set clientParameters(clientParameters: FliptClientParameters) {
+    this._clientParameters = clientParameters;
+  }
+
+  private get clientParameters(): FliptClientParameters {
+    return this._clientParameters;
+  }
+
+  private set namespace(namespace: string) {
+    this._namespace = namespace;
+  }
+
+  private get namespace() {
+    return this._namespace;
+  }
+
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  private set client(client: FliptClient) {
+    this._client = client;
+  }
+
+  private get client(): FliptClient {
+    if (!this._client) {
+      throw new ProviderNotReadyError('Provider is not initialized');
+    }
+    return this._client;
+  }
+
+  async initialize(): Promise<void> {
+    this.client = new FliptClient({
+      url: this.clientParameters.url,
+      authenticationStrategy: this.clientParameters.authenticationStrategy,
+    });
+    this.status = ProviderStatus.READY;
+  }
+
+  async resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<boolean>> {
+    const evalContext: Record<string, string> = transformContext(context);
+
+    try {
+      const booleanEvaluation = await this.client.evaluation.boolean({
+        namespaceKey: this.namespace,
+        flagKey,
+        entityId: context.targetingKey ?? '',
+        context: evalContext,
+      });
+
+      switch (booleanEvaluation?.reason) {
+        case 'DEFAULT_EVALUATION_REASON':
+          return {
+            value: booleanEvaluation?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case 'MATCH_EVALUATION_REASON':
+          return {
+            value: booleanEvaluation?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case 'UNKNOWN_EVALUATION_REASON':
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.UNKNOWN,
+          };
+        default:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.ERROR,
```

Comment:
Should this be considered an error? Perhaps `StandardResolutionReasons.DEFAULT` would be a better fit.